pyproject.toml

```diff
@@ -81,7 +81,7 @@ format = "ruff format {args}"
 format-check = "ruff format --check {args}"
 
 [tool.hatch.envs.test]
-extra-dependencies = [
+dependencies = [
```

Comment:
`test` inherit from `default` but does not need its dependencies (`reno`, `ruff`, ...). 
Using `dependencies` here then allows to use `extra-dependencies` in the `e2e` environment.

Reply:
Ok I've now made `test` a self-referential environment:
- previously, it inherited from `default` and overrode dependencies
- now it does not inherit from `default`; specifies its dependencies

it has a similar effect, but maybe it is clearer